languages/cpp/examples/ExampleUse.md

```diff
@@ -0,0 +1,122 @@
+# Examples
+
+## Prerequisites
+
+### Bitwarden Libraries
+
+Have the two Bitwarden libraries at the same path:
+
+- `BitwardenClient`
+- `bitwarden_c`
+
+For each OS the library files will be the following:
+
+- macOS: `libBitwardenClient.dylib` and `libbitwarden_c.dylib`
+- Linux: `libBitwardenClient.so` and `libbitwarden_c.so`
+- Windows:  `BitwardenClient.dll` and `bitwarden_c.dll`
+
+Follow the [cmake build guide](../CMakeBuild.md) to create the libraries locally.
+
+### Include Directory
+
+`include` directory contains:
+
+- `BitwardenLibrary.h`
+- `BitwardenClient.h`
+- `BitwardenSettings.h`
+- `CommandRunner.h`
+- `Projects.h`
+- `Secrets.h`
+- `schemas.hpp`
+
+### Other Libraries
+
+- `nlohmann-json` (<https://github.com/nlohmann/json>)
+- `boost` (<https://www.boost.org/>)
+
+### Compiling
+
+Use g++/clang++ for compiling.
+
+Example of the folder structure (macOS):
+
+```text
+--root
+    --build
+        `libBitwardenClient.dylib`
+        `libbitwarden_c.dylib`
+    --include
+        --`BitwardenLibrary.h`
+        --`BitwardenClient.h`
+        --`BitwardenSettings.h`
+        --`CommandRunner.h`
+        --`Projects.h`
+        --`Secrets.h`
+        --`schemas.hpp`
+    --examples
+        --`Wrapper.cpp`
+```
+
+Set the environment variable path for the Bitwarden libraries.
+
+For macOS:
+
+```bash
+export DYLD_LIBRARY_PATH=/path/to/your/library:$DYLD_LIBRARY_PATH
+```
+
+For Linux:
+
+```bash
+export LD_LIBRARY_PATH=/path/to/your/library:$LD_LIBRARY_PATH
+```
+
+For Windows:
+
+```shell
+ set "PATH=%PATH%;C:\path\to\your\library"
+```
+
+Set environment variables used in `Wrapper.cpp`:
+
+```bash
+export ACCESS_TOKEN=<"access-token">
+export ORGANIZATION_ID=<"organization-id">
+export API_URL=http://localhost:4000
+export IDENTITY_URL=http://localhost:33656
+```
+
+Compile:
+
+```bash
+cd examples
+clang++ -std=c++20 -I../include -I/path/to/include/nlohmann  -I/path/to/include/boost -L../build/ -o MyBitwardenApp Wrapper.cpp -lBitwardenClient -ldl
+```
+
+for Windows `-ldl` should be excluded,
+
+The result is `MyBitwardenApp` in the `examples` directory, and can be ran from the `examples` directory:
+
+```bash
+./MyBitwardenApp
+```
+
+#### Example
+
+##### macOS
+
+Set environment variables:
+
+```bash
+export DYLD_LIBRARY_PATH=/path/to/your/library:$DYLD_LIBRARY_PATH
+export ACCESS_TOKEN=<"access-token">
+export ORGANIZATION_ID=<"organization-id">
+export API_URL=http://localhost:4000
+export IDENTITY_URL=http://localhost:33656
+```
+
+Compile:
+
+```bash
+clang++ -std=c++20 -I../include -I/opt/homebrew/include -L../build/ -o MyBitwardenApp Wrapper.cpp -lBitwardenClient -ldl
+```
```

Comment:
I am still a bit lost with this one, since it almost does not differ from the steps in `### Compiling`, except for the homebrew include path.
Feels like the `#### Example` does not add anything to what is documented one header before this.

Reply:
When I first went through this, there was a spelling mistake in the homebrew include path in `CMakeBuild.md` and none of the example commands worked. 

It took me awhile to figure out what was actually needed in the `-I/opt/homebrew/include` both in the cmake build documentation and the example use documentation.

What about just adding a note for macOS libraries installed via homebrew use this flag? 
https://github.com/bitwarden/sdk/pull/854/commits/7e6b820dee4d3e9e420eab35131a7d92e4b088ed